File: admin_ui/src/api/professionals.ts
import { api } from './axios'
import type { PhoneNumberJson } from '../utils/phone'

export type Slot = { time: string; status: 'available' | 'booked' | 'blocked' }
export type ProfessionalBrief = { professional_id: string; name: string; employee_id?: string }
export type Professional = { 
  name: string
  professional_id?: string
  employee_id?: string
  price: number; 
  slots: Slot[]; 
  services?: string[];
  phone?: string;
  degree?: string;
  address?: string;
  bio?: string;
}
export type ProfessionalFull = { 
  tenant?: string
  professional_id?: string
  employee_id?: string
  name: string; 
  price?: number; 
  active?: boolean; 
  slots?: Slot[]; 
  services?: string[];
  availability_criteria?: 'daily' | 'weekly' | 'monthly';
  available_days?: number[];
  phone?: string;
  phone_number?: PhoneNumberJson | null;
  degree?: string;
  address?: string;
  bio?: string;
  created_by?: string | null;
  updated_by?: string | null;
}

/** One row per professional; use ``professional_id`` in API paths (names may repeat). */
export async function listProfessionalBriefs(tenant: string): Promise<ProfessionalBrief[]> {
  const res = await api.get(`/tenants/${tenant}/professionals`)
  return res.data as ProfessionalBrief[]
}

export async function getProfessionalSlots(tenant: string, professional: string): Promise<Slot[]> {
  const res = await api.get(`/tenants/${tenant}/professionals/${encodeURIComponent(professional)}/slots`)
  return res.data as Slot[]
}

export async function createProfessional(
  tenant: string,
  payload: { 
    name: string
    employee_id: string
    price?: number
    slot_interval_minutes?: number
    work_start?: string
    work_end?: string
    availability_criteria?: 'daily' | 'weekly' | 'monthly'
    available_days?: number[]
    services?: string[]
    phone?: string
    degree?: string
    address?: string
    bio?: string
  }
): Promise<Professional> {
  const res = await api.post(`/tenants/${tenant}/professionals`, payload)
  return res.data as Professional
}

export async function updateProfessional(
  tenant: string,
  professionalKey: string,
  payload: Partial<ProfessionalFull>
): Promise<ProfessionalFull> {
  const res = await api.patch(`/tenants/${tenant}/professionals/${encodeURIComponent(professionalKey)}`, payload)
  return res.data as ProfessionalFull
}

export async function updateProfessionalSlots(
  tenant: string,
  professional: string,
  slots: Array<string | Slot>,
  date?: string
): Promise<Professional> {
  const res = await api.put(`/tenants/${tenant}/professionals/${encodeURIComponent(professional)}/slots`, { slots, date })
  return res.data as Professional
}

export async function listProfessionalsFull(
  tenant: string,
  params: { active?: boolean } = {}
): Promise<ProfessionalFull[]> {
  const res = await api.get(`/tenants/${tenant}/professionals/full`, { params })
  return res.data as ProfessionalFull[]
}

export async function setProfessionalActive(
  tenant: string,
  professionalKey: string,
  active: boolean
): Promise<ProfessionalFull> {
  const res = await api.patch(`/tenants/${tenant}/professionals/${encodeURIComponent(professionalKey)}`, { active })
  return res.data as ProfessionalFull
}

File: app/models/professionals.py
# app/models/professionals.py
from __future__ import annotations
from typing import List, Optional, Union, Dict, Any
from pydantic import BaseModel, Field, ConfigDict
from app.models.core import Slot


class Professional(BaseModel):
    model_config = ConfigDict(extra="ignore")

    name: str
    professional_id: Optional[str] = None
    employee_id: Optional[str] = None
    short_name: Optional[str] = None
    price: float
    slots: List[Slot] = Field(default_factory=list)
    active: bool = True
    availability_criteria: str = Field(default="daily", description="daily | weekly | monthly")
    available_days: List[int] = Field(default_factory=list)
    services: List[str] = Field(default_factory=list)
    phone: Optional[str] = None
    degree: Optional[str] = None
    address: Optional[str] = None
    bio: Optional[str] = None


class ProfessionalCreate(BaseModel):
    name: str
    employee_id: str
    price: float = 0.0
    slots: Union[List[str], List[Slot]] = Field(default_factory=list)
    slot_interval_minutes: int = Field(default=30, ge=5, le=240)
    work_start: str = Field(default="09:00", pattern=r"^\d{2}:\d{2}$")
    work_end: str = Field(default="18:00", pattern=r"^\d{2}:\d{2}$")
    active: bool = True
    availability_criteria: str = "daily"
    available_days: List[int] = Field(default_factory=list)
    services: List[str] = Field(default_factory=list)
    phone: Optional[str] = None
    degree: Optional[str] = None
    address: Optional[str] = None
    bio: Optional[str] = None


class ProfessionalPatch(BaseModel):
    name: Optional[str] = None
    employee_id: Optional[str] = None
    price: Optional[float] = None
    slots: Optional[Union[List[str], List[Slot]]] = None
    active: Optional[bool] = None
    availability_criteria: Optional[str] = None
    available_days: Optional[List[int]] = None
    services: Optional[List[str]] = None
    phone: Optional[str] = None
    degree: Optional[str] = None
    address: Optional[str] = None
    bio: Optional[str] = None


class ProfessionalListResponse(BaseModel):
    items: List[Dict[str, Any]]
    total: int
    page: int


class ProfessionalOut(BaseModel):
    name: str
    price: float
    slots: List[Slot]
    active: bool
    availability_criteria: Optional[str] = None
    available_days: Optional[List[int]] = None
    services: Optional[List[str]] = None
    phone: Optional[str] = None
    degree: Optional[str] = None
    address: Optional[str] = None
    bio: Optional[str] = None


class UpdateSlotsBody(BaseModel):
    slots: List[Union[str, Slot]] = Field(default_factory=list)
    date: Optional[str] = None


class SlotStatusPatch(BaseModel):
    status: str = Field(..., pattern="^(available|blocked)$")
    size: int
